README.md

```diff
@@ -230,3 +277,51 @@ const Counter = () => {
   );
 };
 ```
+
+### Migration from v1 to v2
+
+All value based hooks like `useBoolean`, `useNumber` etc. Are changed to 
+be using arrays, since it's more safe for reference equality, and also 
+makes it easier to use many `useSmth` without renaming `value` in destructuring.
+
+So if you had 
+```javascript
+const { value: showHeader, ...showHeaderActions } = useBoolean(true)
+const { value: showFooter, ...setShowFooterActions } = useBoolean(true)
+```
+It will become
+```javascript
+const [showHeader, showHeaderActions] = useBoolean(true)
+const [showFooter, showFooterActions] = useBoolean(true)
+```
+
+Note that despite this code seems to be looking the same, it's not. Cause `showHeaderActions` in v1 will result
+in new object reference every rerender. While in v2 actions are memoized using `useMemo` and their reference will not
+change.
+It enables us passing `actions` down the props without useless re-renders, also it prevents `useEffects` and
+other hooks from re-run/new reference if autofix of ESLint rule `react-hooks/extraneous-deps` will add them as dependencies.
+
+### useInput migration
```

Comment:
@kitze please take a look at this. Does it make sense to merge it together? Or this composition is ok?
Justifications for these changes are described above.